core/src/scene/importer.cpp

```diff
@@ -0,0 +1,257 @@
+#include "importer.h"
+#include "platform.h"
+#include "scene/sceneLoader.h"
+#include "util/topologicalSort.h"
+#include "yaml-cpp/yaml.h"
+
+#include <regex>
+
+using YAML::Node;
+using YAML::NodeType;
+
+namespace Tangram {
+
+Node Importer::applySceneImports(const std::string& scenePath) {
+
+    if (loadScene(scenePath)) {
+        return importScenes(scenePath);
+    }
+
+    return Node();
+}
+
+std::string Importer::getFilename(const std::string& filePath) {
+    std::string sceneName = filePath;
+    std::regex r("[^//]+$");
+    std::smatch match;
+    if (std::regex_search(sceneName, match, r)) { sceneName = match[0]; }
+    return sceneName;
+}
+
+std::string Importer::normalizePath(const std::string &path,
+                                         const std::string &parentPath) {
+
+    //TODO: Handle network urls
+
+    if (path[0] == '/') {
+        //absolute path, return as it is
+        return path;
+    }
+
+    std::regex r("[^//]+$");
+    return std::regex_replace(parentPath, r, path);
+}
+
+void Importer::normalizeSceneImports(Node& root, const std::string& parentPath) {
+    if (Node import = root["import"]) {
+        if (import.IsScalar()) {
+            import = normalizePath(import.Scalar(), parentPath);
+        } else if (import.IsSequence()) {
+            for (Node imp : import) {
+                if (imp.IsScalar()) { imp = normalizePath(imp.Scalar(), parentPath); }
+            }
+        }
+    }
+}
+
+void Importer::setNormalizedTexture(Node& texture, const std::vector<std::string>& names,
+        const std::string& parentPath) {
+
+    for (size_t index = 0; index < names.size(); index++) {
+        auto& name = names[index];
+        std::string normTexPath;
+
+        //get normalized texture path
+        if (m_textureNames.find(name) == m_textureNames.end()) {
+            normTexPath = normalizePath(name, parentPath);
+            m_textureNames[name] = normTexPath;
+        } else {
+            normTexPath = m_textureNames.at(name);
+        }
+
+        //set yaml node with normalized texture path
+        if (names.size() > 1) {
+            texture[index] = normTexPath;
+        } else {
+            texture = normTexPath;
+        }
+    }
+}
+
+void Importer::normalizeSceneTextures(Node& root, const std::string& parentPath) {
+
+    if (Node textures = root["textures"]) {
+        for (auto texture : textures) {
+            if (Node textureUrl = texture.second["url"]) {
+                setNormalizedTexture(textureUrl, {textureUrl.Scalar()}, parentPath);
+            }
+        }
+    }
+
+    if (Node styles = root["styles"]) {
+
+        for (auto styleNode : styles) {
+
+            auto style = styleNode.second;
+            //style->texture
+            if (Node texture = style["texture"]) {
+                if (texture.IsScalar()) {
+                    setNormalizedTexture(texture, {texture.Scalar()}, parentPath);
```

Comment:
Good you pointed this out. I missed it before.Will handle this in following commits. Thanks